tests/lib/data_fifo/src/main.c

```diff
@@ -31,6 +31,21 @@ static void internal_test_remaining_elements(struct data_fifo *data_fifo, uint32
 		      num_locked, line);
 }
 
+ZTEST(suite_data_fifo, test_data_fifo_deinit_ok)
```

Comment:
Nice.

Suggest we add a bit more substantial test where we init a fifo, fill it, deinit it, init it again and fill it.

Reply:
Ok will add something.